core/src/main/java/io/aiven/klaw/service/MailUtils.java

```diff
@@ -343,10 +344,10 @@ private void sendMail(
             }
 
             try {
-              emailIdTeam = dbHandle.selectAllTeamsOfUsers(username, tenantId).get(0).getTeammail();
+              List<Team> allTeams = dbHandle.selectAllTeamsOfUsers(username, tenantId);
+              if (!allTeams.isEmpty()) emailIdTeam = allTeams.get(0).getTeammail();
             } catch (Exception e) {
-              log.error("Exception:", e);
-              emailIdTeam = null;
+              log.error("Exception :", e);
```

Comment:
Maybe more of a style thing but adding the method name inside of here might make it easier to track it back to this location when it fails. e.g. log.error("sendMail() Exception:",e) or if you want to be more specific as its in a nested try log.error("sendMail.selectAllTeamsOfUsers() Exception:",e) as inside the completable future I think the logger might lose some context.

Reply:
Yeah I am not 100% sure but I think in the completable Future the parent Thread context is lost. I would have test it to be sure. so I don't think you would get a full stack trace.

Reply:
agree, for 'Future' threads, that context is lost. we can improve this.